app/pktgen-stats.c

```diff
@@ -221,23 +221,30 @@ void
 pktgen_get_link_status(port_info_t *pinfo)
 {
     struct rte_eth_link link = {0};
-    uint32_t link_speed      = pinfo->link.link_speed;
+    uint32_t prev_link_speed = pinfo->link.link_speed;
 
     if (rte_eth_link_get_nowait(pinfo->pid, &link) == 0) {
-        if (link.link_speed == RTE_ETH_SPEED_NUM_UNKNOWN) {
-            /* Setup a few default values to prevent problems later. */
-            link.link_speed   = RTE_ETH_SPEED_NUM_10G;
-            link.link_duplex  = RTE_ETH_LINK_FULL_DUPLEX;
-            link.link_autoneg = RTE_ETH_LINK_SPEED_AUTONEG;
-            link.link_status  = RTE_ETH_LINK_UP;
-        }
-        if (link.link_status == RTE_ETH_LINK_UP && link_speed != link.link_speed) {
-            pinfo->link.link_status  = link.link_status;
-            pinfo->link.link_speed   = link.link_speed;
-            pinfo->link.link_autoneg = link.link_autoneg;
-            pinfo->link.link_duplex  = link.link_duplex;
+        /* Update cached link info whenever any field changes */
+        if (link.link_status != pinfo->link.link_status ||
```

Comment:
Now that I see the code, it may have been better to use memcmp() and memcpy() to update the pinfo->link instead of the assigning each field.
We do not need to change the code to use these routines, but it is most likely clearer this way.

Reply:
got it, resolved